cmds/mv/mv.go

```diff
@@ -20,23 +20,57 @@ import (
 	"path/filepath"
 )
 
+var (
+	update = flag.Bool("u", false, "move only when the SOURCE file is newer than the destination file or when the destination file is missing")
+)
+
 func usage() {
-	fmt.Fprintf(os.Stderr, "usage: %s source target\n", os.Args[0])
-	fmt.Fprintf(os.Stderr, "       %s source ... directory\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "Usage: %s [ARGS] source target\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "       %s [ARGS] source ... directory\n", os.Args[0])
+	flag.PrintDefaults()
 	os.Exit(1)
 }
 
+func moveFile(source string, dest string) error {
+	if *update {
+		var err error
+		var sourceModTime, destModTime os.FileInfo
+
+		sourceModTime, err = os.Lstat(source)
+		if err != nil {
+			return err
+		}
+		var sourceDate = sourceModTime.ModTime()
+
+		// Check if the destination already exists and was touched later than the source
+		destModTime, err = os.Lstat(dest)
+		if err == nil {
```

Comment:
super nitpicky, but: Go style would be not to nest further, and also to make the most likely path to happen the straight-forward one:

```go
sourceInfo, err := os.Lstat(source)
if err != nil {
  return err
}
destInfo, err := os.Lstat(dest)
if err != nil {
  return err
}
if destInfo.ModTime().After(sourceInfo.ModTime()) {
  return nil
}
```

also, you can use := to avoid having to declare variables here.

I'll mark this as optional.

Reply:
Alright. So it's fine to use the same variable (e.g. err) for multiple purposes one after another?
That's why I introduced the additional scope in TestMvUpdate - so the variable isn't reused but we have the same name in different scopes.